src/kvirc/sparser/KviIrcServerParser_literalHandlers.cpp

```diff
@@ -1002,15 +1006,16 @@ void KviIrcServerParser::parseLiteralPrivmsg(KviIrcMessage * msg)
 				       console, szNick, szUser, szHost, szMsg, msg->messageTagsKvsHash()))
 				{
 					// check if the scripter hasn't created it
-					query = msg->connection()->findQuery(szNick);
+					query = msg->connection()->findQuery(szWindow);
 				}
 				else
 				{
 					// no query yet, create it!
 					// this will trigger OnQueryWindowCreated
-					query = console->connection()->createQuery(szNick);
+					query = console->connection()->createQuery(szWindow);
 					// and this will trigger OnQueryTargetAdded
-					query->setTarget(szNick, szUser, szHost);
+					if (!bSelfMessage)
+						query->setTarget(szNick, szUser, szHost);
```

Comment:
Docs for that function say it supports `*` in such case. Like when you do `/query nick`